src/funnels/automation/StableSwapper.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: © 2020 Lev Livnev <lev@liv.nev.org.uk>
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface SwapperLike {
+    function swap(address src, address dst, uint256 amt, uint256 minOut, address callee, bytes calldata data) external returns (uint256 out);
+}
+
+contract StableSwapper {
+    mapping (address => uint256) public wards;                           // admin
+    mapping (address => uint256) public buds;                            // facilitators
```

Comment:
If this is the contract that facilitators can use, then here we could remove buds and just have roles.
In contraposition, if Swapper is just a middleman contract that will be only set by subdao proxy, roles should probably be removed there.

Reply:
Yeah that's the contract that facilitators configure and for which they set keepers. I'm not against replacing `buds` with using roles but part of me feels that this is a bit overkill as the only `buds` here should be the facilitator multisig (and not any automation contract, unlike in the layer below).

Reply:
I'm more thinking on setting a standardize way to manage the facilitators/operators and not have multiple structures. If we use role we can list all the facilitators and which actions they have access.
I know bots structure breaks it a bit, I almost feel we should not have bots and just delegate that to un upper level contract, but yeah we keep adding waste of gas.

Reply:
So from what we've discussed, having `roles` in the primitive `Swapper` makes sense because that decreases the risk on adding automation contracts. In theory subdao proxy could add an automation contract via roles and being sure than more than the `cap` per `hop` can't be at risk. If we use `wards`, a malicious automation contract could also potentially change `cap` and `hop`. 

On the other hand, if automation contracts are seeing as just pieces owned by the facilitators and not really sub dao contracts, then I'd agree that `buds` or `roles` shouldn't be used for them, instead directly `wards`.

Reply:
Yeah as discussed I think we want to avoid needing to audit each automation contract to make sure it doesn't change the rate limiting, so separating to a different role makes sense there.

IMO the automation contracts (1) should not be forced to be owned by subdao proxies, but (2) can be in practice. So we shouldn't use `roles` there because of (1) but should have at least `bus` because of (2).

Reply:
Automation contracts now uses wards and buds (keepers) only at https://github.com/makerdao/dss-allocator/pull/15.
@sunbreak1211 can we resolve this?